### ISSUE TITLE
Focus ancestry gets announced in Firefox

### ISSUE THREAD

**Issue by Neurrone** [nvaccess/nvda#11323]
CC: @jcsteh 

Originally reported in #11305.

### Steps to reproduce:

1. Visit [Mdn's page on the select element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/select)
2. Press c to move to the select example. Interact with it and change the currently selected option.

### Actual behavior:

Here's some speech output where I am in the select element and choose a new option.

```
Choose a pet:  combo box  Hamster  collapsed
combo box  expanded
Hamster  list
Hamster  4 of 7
Parrot  5 of 7
<select> - HTML: Hypertext Markup Language | MDN  document
main landmark
article
MDN Web Docs Interactive Example  frame
MDN Web Docs Interactive Example  document
Choose a pet:  combo box  Parrot  collapsed
```

### Expected behavior:

The extra 5 lines of focus ancestry shouldn't be announced

#### NVDA installed/portable/running from source:
Installed

#### NVDA version:
alpha-20432,d5207132

#### Windows version:
Windows 10 2004

#### Name and version of other software in use when reproducing the issue:
Firefox 79.0 B2
Can't be reproduced with Chrome so I suspect this is a Firefox issue

#### Does the issue still occur after restarting your computer?
Yes

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
No
#### If addons are disabled, is your problem still occuring?
Yes

#### Did you try to run the COM registry fixing tool in NVDA menu / tools?
No

**Comment by jcsteh** [comment]
>     2. Press c to move to the select example. Interact with it and change the currently selected option.

At risk of sounding pedantic, how exactly are you "interacting" with it? Are you pressing enter, then arrow keys? or are you pressing alt+downArrow then arrow keys?

The reason I ask is that I can't reproduce this, so I'm wondering whether I'm doing something different. Also, this:

> combo box  expanded

suggests that the combo box is expanding, which doesn't usually happen unless you press alt+downArrow. But even if I do press alt+downArrow and get this message, I still can't reproduce it.

> Choose a pet:  combo box  Parrot  collapsed

Here, the combo box collapses again. What did you press here? Was this just after pressing arrow keys or did you press enter or alt+upArrow?

In short, it'd be good to know the exact keystrokes you pressed so I can reproduce this precisely. The tiniest detail might matter here.

**Comment by Neurrone** [comment]
@jcsteh that's a good point. What I did was:

1. Press c
2. Press enter to go to focus mode
3. Press alt+down to expand the list
4. Press up/down to change options
5. Press enter

Interestingly if I omit 2, I don't encounter this issue. Likewise if I don't press alt+down to expand the list first.

**Comment by jcsteh** [comment]
Just to clarify, you only see this when you press enter, not when you press the up/down arrow keys?

That's a known bug. See https://bugzilla.mozilla.org/show_bug.cgi?id=1422176. However, it was closed because the usual patterns for interacting with these controls (omitting either step or step 3 as you noted) doesn't cause this problem, and no one seemed to use or care about this edge case. It seems you are proof that this isn't true :), so I'll reopen that bug. That said, it's still low priority given the difficulty of fixing it and the fact that there is a simple workaround.

**Comment by jcsteh** [comment]
Actually, I think this needs a new Mozilla bug. Originally, this was a lot worse in NVDA because it caused focus mode switching to break if you didn't press enter before pressing alt+downArrow. That is no longer the case, so now we're just dealing with the extra verbosity. Same underlying cause, but very different user impact.

You can file a new Mozilla bug if you wish or I will file it when I get a chance (I'm going away for a few days shortly).

**Comment by Neurrone** [comment]
Np, this isn't urgent.

Honestly the only way I even noticed this was while performing a workshop for some colleagues on web accessibility. I happened to be mentioning that pressing alt+down with a keyboard allows you to expand it and this keystroke isn't SR specific. Then when I selected an option, I noticed this happening.

**Comment by Neurrone** [comment]
Ugh, stupid github's ordering of the buttons, whoever thought it was a good idea to have close and comment go before comment wth

**Comment by XLTechie** [comment]
Please consider updating the title of this to indicate exactly what corner case 
thing it is that causes the Focus ancestry to get announced, so future issue 
searchers won't have to read this far to know if it's a duplicate.

Thanks

**Comment by Adriani90** [comment]
Did anyone create a Mozilla bug for this already?

**Comment by jcsteh** [comment]
It's covered by https://bugzilla.mozilla.org/show_bug.cgi?id=1549202

**Comment by jcsteh** [comment]
This was fixed in Firefox.
